### ISSUE TITLE
Why is the accuracy of the llava-video-hf model only 40% in the two benchmarks longvideo and videomme after custom code inference, which is far from the 61.9 and 64.6 in the lmms-eval evaluation data in the paper? I don't know the reason，below is the code

### ISSUE THREAD

**Issue by zapqqqwe** [ncTimTang/AKS#12]
import os
import torch
os.environ['CUDA_LAUNCH_BLOCKING'] = "1"
from transformers import AutoProcessor, LlavaForConditionalGeneration
# LlavaNextVideoForConditionalGeneration
# model_path="/mnt/bn/material-test/lichenglin/models/LLaVA-NeXT-Video-7B-hf"
model_path="/mnt/bn/material-test/lichenglin/models/LLaVA-Video-7B-Qwen2-hf"
model =LlavaForConditionalGeneration.from_pretrained(model_path, device_map="auto")
processor = AutoProcessor.from_pretrained(model_path)

import json

from tqdm import tqdm
import os
from decord import VideoReader, cpu
import numpy as np
json_file="/root/frame_pruning/AKS/datasets/longvideobench/include_frame_idx.json"
video_path="/mnt/bn/material-test/dataset/LongVideoBench/videos"
output_file="./output_longvideo_llava_video_qw2.jsonl"
with open(json_file,'r') as f:
    datas=json.load(f)
def load_video_index(video_path, max_frames_num, doc):
    if max_frames_num == 0:
        return np.zeros((1, 336, 336, 3))
    vr = VideoReader(video_path, ctx=cpu(0), num_threads=1)
    total_frame_num = len(vr)
    video_time = total_frame_num / vr.get_avg_fps()
    fps = round(vr.get_avg_fps())
    top_id = doc['frame_idx']
    frame_idx = top_id[:max_frames_num]
    frame_idx = sorted(frame_idx)
    frame_time = [i / fps for i in frame_idx]
    if len(frame_idx) < max_frames_num:
        sample_fps = max_frames_num
        uniform_sampled_frames = np.linspace(0, total_frame_num - 1, sample_fps, dtype=int)
        frame_idx = uniform_sampled_frames.tolist()
        frame_time = [i / vr.get_avg_fps() for i in frame_idx]
    frame_time = ",".join([f"{i:.2f}s" for i in frame_time])
    spare_frames = vr.get_batch(frame_idx).asnumpy()
    # import pdb;pdb.set_trace()
    return spare_frames, frame_time, video_time
frames_num=32

with open(json_file,'r') as f:
    datas=json.load(f)
post_prompt="\nAnswer with the option's letter from the given choices directly. "

for data in tqdm(datas):
    v_path=os.path.join(video_path,data['video_path'])
    prompt_=post_prompt+data['question']+'\n'+"\n".join([". ".join([chr(ord("A") + i), candidate]) for i, candidate in enumerate(data['candidates'])])
    prompt = f"USER: <video>\n{prompt_} ASSISTANT:"
    spare_frames, frame_time, video_time=load_video_index(v_path,max_frames_num=frames_num,doc=data)
    inputs_video = processor(text=prompt, videos=spare_frames, return_tensors="pt").to(model.device)
    generate_ids = model.generate(**inputs_video,max_new_tokens=32,temperature=0,do_sample=False)
    data['output']=processor.batch_decode(generate_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)[0]
with open(output_file,'w') as f:
    json.dump(datas,f,indent=4,ensure_ascii=False)

**Comment by zapqqqwe** [comment]
the model we use is https://huggingface.co/SanyaChoi/LLaVA-Video-7B-Qwen2-hf

**Comment by ncTimTang** [comment]
Hi @zapqqqwe ,

We use the template [here](https://github.com/LLaVA-VL/LLaVA-NeXT/blob/main/llava/conversation.py#L543) rather than 

```Shell
USER: <video>\n{prompt_} ASSISTANT:
```

Since we haven't used this [model](https://huggingface.co/SanyaChoi/LLaVA-Video-7B-Qwen2-hf) before, we're not sure if that's the reason. You can try replacing the template and retest. If there are still questions, please feel free to discuss.

Best,

Tim
